qlty-analysis/src/code/visitor.rs

```diff
@@ -1,6 +1,9 @@
 use crate::Language;
+use tracing::warn;
 use tree_sitter::{Node, TreeCursor};
 
+const MAX_CURSOR_DEPTH: u32 = 400;
```

Comment:
How did you select 400?

Reply:
Basically ran `qlty smells --all` a number of times on a folder containing these files and two others while logging the cursor depth, usually stack overflow occurred above 450 depth.

 We can lower it to be safer.

Reply:
I think 400 is fine.

Reply:
I think going to lower it a bit down to 300 to be safer on more systems.